DOC-6565: Added Running the TVU section (#2242)
corda/corda-docs-portal: content/en/platform/corda/4.12/enterprise/node/operating/tvu/running-tvu.md

---
description: "The steps that TVU goes through when verifying pre-4.12 database of transactions."
date: '2024-11-04'
menu:
  corda-enterprise-4-12:
    identifier: corda-enterprise-4-12-running-tvu
    parent: corda-enterprise-4-12-tvu
tags:
- tvu
- transaction validator utility
title: Running the TVU
weight: 150
---

# Running the TVU

> This section describes the requirements to run the Transaction Validator Utility (TVU) to verify a pre-4.12 database of transactions. Please read this section in conjunction with [Corda Enterprise Edition 4.11 to 4.12 upgrade guide]({{< relref "../../../upgrade-guide.md" >}}) which describes in more detail the upgrade process.

Reviewer: ```suggestion
This section describes the requirements to run the Transaction Validator Utility (TVU) to verify a pre-4.12 database of transactions. Please read this section in conjunction with [Corda Enterprise Edition 4.11 to 4.12 upgrade guide]({{< relref "../../../upgrade-guide.md" >}}) which describes the upgrade process in more detail.
```

> The TVU is compatible only with 4.11 database schemas. To verify transactions on a pre-4.11 database, you must first upgrade the database to version 4.11. You must run the TVU in the same environment as a 4.12 node, in terms of `cordapps` and `legacy-jar` folders, with the exception that the database it connects to is the 4.11 database being checked. Note that you do not need to create a `legacy-contracts` folder when running the TVU. The TVU will extract what it needs from the database. Before performing the following steps, ensure everything is backed up, as changes will be made to folders as described below.

Reviewer 2: should be legacy-jars 

i.e. need to add in the 's'

Author: updated

1. Place the TVU JAR in your Corda 4.11 node directory.
2. In the `cordapps` directory, remove the existing JARs and replace them with the 4.12 JDK17/Kotlin 1.9 equivalent JARs.

> 3. If needed, create a `legacy-jars` directory and place into this directory third-party dependencies and any JARs you were dependent upon located in the `drivers` directory. See [Corda Enterprise Edition 4.11 to 4.12 upgrade guide]({{< relref "../../../upgrade-guide.md" >}}) for details.

Reviewer: ```suggestion
3. If needed, create a `legacy-jars` directory, then copy the third-party dependencies and any JARs you were dependent upon from the `drivers` directory to the `legacy-jars` directory . See [Corda Enterprise Edition 4.11 to 4.12 upgrade guide]({{< relref "../../../upgrade-guide.md" >}}) for details.
```

---
description: "The steps that TVU goes through when verifying pre-4.12 database of transactions."
date: '2024-11-04'
menu:
  corda-enterprise-4-12:
    identifier: corda-enterprise-4-12-running-tvu
    parent: corda-enterprise-4-12-tvu
tags:
- tvu
- transaction validator utility
title: Running the TVU
weight: 150
---

# Running the TVU

This section describes the requirements to run the Transaction Validator Utility (TVU) to verify a pre-4.12 database of transactions. Please read this section in conjunction with [Corda Enterprise Edition 4.11 to 4.12 upgrade guide]({{< relref "../../../upgrade-guide.md" >}}) which describes in more detail the upgrade process.

> The TVU is compatible only with 4.11 database schemas. To verify transactions on a pre-4.11 database, you must first upgrade the database to version 4.11. You must run the TVU in the same environment as a 4.12 node, in terms of `cordapps`, `legacy-contract`, and `legacy-jar` folders, except that the database it connects to is the 4.11 database being checked. Before performing the following steps, ensure everything is backed up, as changes will be made to folders as described below.

Reviewer 2: Can you replace this paragraph with the following please,

The TVU is compatible only with 4.11 database schemas. To verify transactions on a pre-4.11 database, you must first upgrade the database to version 4.11. You must run the TVU in the same environment as a 4.12 node, in terms of `cordapps`, and `legacy-jar` folders, with the exception that the database it connects to is the 4.11 database being checked. Note that you do not need to create a legacy-contracts folder when running the TVU. The TVU will extract what it needs from the database. Before performing the following steps, ensure everything is backed up, as changes will be made to folders as described below.

Author: Replaced

1. Place the TVU JAR in your Corda 4.11 node directory.
2. In the `cordapps` directory, remove the existing JARs and replace them with the 4.12 JDK17/Kotlin 1.9 equivalent JARs.

> 3. Create a `legacy-contracts` directory that contains the 4.11 contract JARs (the ones that were previously in the `cordapps` directory).

Reviewer 2: Can you remove point 3 please

Author: Removed
